src/service/cache/value-cache.service.js

```diff
@@ -0,0 +1,447 @@
+const fs = require('node:fs/promises')
+const path = require('node:path')
+
+const { nanoid } = require('nanoid')
+
+const { createFolder } = require('../utils')
+const DeferredPromise = require('../deferred-promise')
+
+const BUFFER_MAX = 250
+const BUFFER_TIMEOUT = 300
+const RESEND_IMMEDIATELY_TIMEOUT = 100
+
+const VALUE_FOLDER = 'values'
+const ERROR_FOLDER = 'values-errors'
+
+const BUFFER_FILE_NAME = 'buffer.tmp'
+
+/**
+ * Local cache implementation to group events and store them when the communication with the North is down.
+ */
+class ValueCacheService {
+  /**
+   * @param {String} northId - The North ID connector
+   * @param {Logger} logger - The logger
+   * @param {String} baseFolder - The North cache folder generated as north-connectorId. This base folder can
+   * be in data-stream or history-query folder depending on the connector use case
+   * @param {Function} northSendValuesCallback - Method used by the North to handle values
+   * @param {Function} northShouldRetryCallback - Method used by the North to retry or not the sending
+   * @param {Object} settings - Cache settings
+   * @return {void}
+   */
+  constructor(
+    northId,
+    logger,
+    baseFolder,
+    northSendValuesCallback,
+    northShouldRetryCallback,
+    settings,
+  ) {
+    this.northId = northId
+    this.logger = logger
+    this.baseFolder = baseFolder
+    this.northSendValuesCallback = northSendValuesCallback
+    this.northShouldRetryCallback = northShouldRetryCallback
+    this.settings = settings
+
+    this.flushBuffer = []
+    this.bufferTimeout = null
+    this.valuesTimeout = null
+    this.queue = new Map() // key: queue filename (randomId.queue.tmp, value: the values in the queue file
+    this.compactedQueue = [] // List of object {fileName, creationAt, numberOfValues} fileName: compact filename (randomId.compact.tmp)
+    this.sendingValuesInProgress = false
+    this.sendNextImmediately = false
+    this.valuesRetryCount = 0
+    this.sendingValues$ = null
+
+    this.valueFolder = path.resolve(this.baseFolder, VALUE_FOLDER)
+    this.errorFolder = path.resolve(this.baseFolder, ERROR_FOLDER)
+  }
+
+  /**
+   * Create databases and folders
+   * @returns {Promise<void>} - The result promise
+   */
```

Comment:
a synthetic reminder of what is done at startup and a reference to the readme would be quite helpful